components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -20,6 +20,7 @@
 #include "base/system/sys_info.h"
 #include "base/task/post_task.h"
 #include "base/values.h"
+#include "brave/browser/brave_stats_updater_util.h"
```

Comment:
Layer violation - component should avoid using chrome(brave) layer.

Reply:
Should I instead add a stats.h and stats.c to brave/common then? Or copy the function?

Reply:
I think moving it to components seems good.
If it's in component layer, chrome layer and other components can use it.
How about moving that api to this brave_referrals component?

Reply:
and which name is more proper between `brave_referrals_api_key` and `brave_stats_api_key`?
I'm not sure which one is more appropriate because it's used from referrals service and stats updater.
Or other neutral name?

Reply:
I think `brave_stats_api_key` makes more sense because it's being used by the stats team. Stats manages both referrals and the usage pings in brave_stats_updater.

Reply:
For the layer violation, it is a little awkward to make these functions just part of referrals. Those platform identifiers are used by both referrals and the stats updater. They are more shared utility functions. Notice that the p3a component also includes `brave_stats_updater_util.h`. Is there anything better we can do here? Putting this stuff in referrals doesn't feel right

Reply:
Ok - how about passing key to `BraveReferralsService` as its ctor's argument?